pkgs/development/compilers/scala/dotty.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchurl, makeWrapper, jre }:
+
+stdenv.mkDerivation rec {
+  version = "0.4.0-RC1";
+  name = "dotty-${version}";
+
+  src = fetchurl {
+    url = "https://github.com/lampepfl/dotty/releases/download/${version}/${name}.tar.gz";
+    sha256 = "1d1ab08b85bd6898ce6273fa50818de0d314fc6e5377fb6ee05494827043321b";
+  };
+
+  propagatedBuildInputs = [ jre ] ;
+  buildInputs = [ makeWrapper ] ;
+
+  installPhase = ''
+    mkdir -p $out
+    mv * $out
+
+    for p in $out/bin/* ; do
```

Comment:
Nitpick: this fits better in `fixupPhase`, and you could write this as `find` expression:

```
  fixupPhase = ''
    for p in $(find $out/bin/ -mindepth 1 -not -name common); do
      wrapProgram $p --set JAVA_HOME ${jre}
    done
  '';
```

Better yet, you can only patch `$out/bin/common`, as it's the only file that uses `JAVA_HOME`:

```
  fixupPhase = ''
    wrapProgram $out/bin/common --set JAVA_HOME ${jre}
  '';
```